<!DOCTYPE html>
<html lang="en-us">
  <head>
    <meta charset="UTF-8">
    <title>Ld ne by geneticswithme</title>
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <link rel="stylesheet" type="text/css" href="stylesheets/normalize.css" media="screen">
    <link href='https://fonts.googleapis.com/css?family=Open+Sans:400,700' rel='stylesheet' type='text/css'>
    <link rel="stylesheet" type="text/css" href="stylesheets/stylesheet.css" media="screen">
    <link rel="stylesheet" type="text/css" href="stylesheets/github-light.css" media="screen">
    <script type="text/javascript"
      src="http://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML">
    </script>
  </head>
  <body>
    <section class="page-header">
      <h1 class="project-name">Ld Ne</h1>
      <h2 class="project-tagline">Estimate recombination rate from LD and the known map length of the chromosomes, estimate Ne based on recombination rate</h2>
      <a href="https://github.com/geneticswithme/LD_Ne" class="btn">View on GitHub</a>
      <a href="https://github.com/geneticswithme/LD_Ne/zipball/master" class="btn">Download .zip</a>
      <a href="https://github.com/geneticswithme/LD_Ne/tarball/master" class="btn">Download .tar.gz</a>
    </section>

    <section class="main-content">
      <p>Estimate recombination rate from LD and the known map length of the chromosomes, estimate Ne based on recombination rate</p>
      <p>This is an example of how to estimate recombination rate(c) in sheep genome with SNPChip data. And then, estimate the effective population size(Ne) based on the recombination rate.</p>
      <p>We conduct estimation of recombination rate using the method described by Tenesa et. al(2007). The main thoughts of this method is based on a famouse equations describing the relationship of Ne and E(r2).</p>
      $$
      E(r^2) \approx \frac{1}{\alpha + kN_ec}
      $$
      Taking both experimental and evolutionary sampling effects into account, the relationship between LD and Ne.
      <span >\[E(r^2)=\frac{1}{\alpha+kN_ec}+\frac{1}{n}\]</span>

      <h2><b>Tutorial</b></h2>

      <h3>1 Calculate LD using Haploview</h3>
      <code>   
      plink --bfile example --recode HV 
      </code>
      <p>This command will generate a serie of input files for Haploview.  
        <br>plink.chr-1.ped, plink.chr-1.info; plink.chr-2.ped, plink.chr-2.info; ...</br></p>

      <p> 
        <code type="bash">
          java -jar Haploview.jar -pedfile plink.chr-1.ped -info plink.chr-1.info -dprime </code>
        <br><font size="3">output file <b>plink.chr-1.ped.LD</b> has columns 
        [1] SNP1_name [2] SNP2_name [3] D. [4] LOD [5] r.2 [6] CIlow [7] CIhi [8] Dist [9] T.list</font>
        <br>
      </p>

      <h3>2 import R functions</h3>
      <p>open R console</p>
<pre><code class="R">source("LD_Ne.R")</code></pre>
      

<h3>3 prepare input dataframe</h3>
<br>map.txt has columns [1]CHR [2]SNP_name [3]BP</br>
<br>cm_map.txt recodes chromosome length of linkage map and physical map. it has columns [1]CHR [2]linkage length of chr [3]physical length of chr</br>

<pre><code>input="plink.chr-23.ped.LD"
input_LD<-read.table(input,header=T,sep="\t")
physical_map<-read.table("SNP.map")
linkage_map<-read.table("sheep_chr_cM.txt",header=T)
L1_pos<-physical_map[match(input_LD[,"L1"],physical_map[,2]),4]
L2_pos<-physical_map[match(input_LD[,"L2"],physical_map[,2]),4]
LD_matrix<-cbind(input_LD[,1:2],input_LD[,5]-1/n,L1_pos,L2_pos,input_LD[,8])
LD_df=LD_matrix
LD_df[,6]<-LD_matrix[,6]/1000000
colnames(LD_df)<-c("L1","L2","r2","L1_pos","L2_pos","dist")</code></pre>
<p><b>LD_matrix</b> has columns [1] SNP1_name [2] SNP2_name [3] r.2 [4] SNP1_pos [5] SNP2_pos [6]Dist in Mb
</p>

<h3>4 calculate average beta on the whole chromosome</h3>
<code>entire_chr_beta=beta_estimate(LD_df,alpha=1)
entire_chr_beta</code>
<h3>5 Calculate recombinate rate along chromosome with windown size=500kb, overlap=1/2 * window_size</h3>    
<code>out_c<-calculate_c_for_each_window(LD_df,window_size=500000,alpha=1,linkage_map=linkage_map,chr=23)
c_map<-recombination_rate_accumulated_map(out_c)</code>

<h3>6 Plot recombination rate</h3>
<pre><code>chr=23
chr_genetic_L=linkage_map[chr,2]
chr_physical_L=linkage_map[chr,1]
plot(x=c_map[,1],y=c_map[,2],main=paste0("chr",chr),axes=FALSE,xlab="chromosome length(Mb)",ylab="cM",type="l")
y_axis=seq(0,chr_genetic_L,2)
axis(2,at=y_axis)
x_axis=seq(0,chr_physical_L*1000000,5000000)
axis(1,at=x_axis,labels=paste0(x_axis/1000000) )</code></pre>

      <footer class="site-footer">
        <span class="site-footer-owner"><a href="https://github.com/geneticswithme/LD_Ne">Ld ne</a> is maintained by <a href="https://github.com/geneticswithme">geneticswithme</a>.</span>

        <span class="site-footer-credits">This page was generated by <a href="https://pages.github.com">GitHub Pages</a> using the <a href="https://github.com/jasonlong/cayman-theme">Cayman theme</a> by <a href="https://twitter.com/jasonlong">Jason Long</a>.</span>
      </footer>

    </section>

  
  </body>
</html>
